<a href="https://colab.research.google.com/github/OmdenaAI/omdena-canada-aigovernance/blob/main/coronacoviedpolicy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [526]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as se
import numpy as np
file=pd.read_csv('/content/drive/MyDrive/coronanet_release_Canada.csv')
import datetime as dt


Cleaning Data

# First we read the data

In [527]:
#file.describe

In [528]:
#file.head()


We have 40 features ,lets go one by one to differentiate between categorical and text

We will add three columns showing the effect of policies to food availability ,those three columns indicate how the effect has been.
the value so fare is 0 till we decided to fill it 

In [529]:
file['international exchange']=0
file['Farm financial stability']=0
file['food_Transportation']=0

Since the tarrget country is Canda ,so no need for this column any more ,we can delete it .also delete 'target_region','target_province','target_city' becouse this columns full of null,and delete the link 

In [530]:
file.drop(columns=file[['target_country','target_region','target_province','target_city']],inplace=True)

In [531]:
file.drop(columns=file[['link',]],inplace=True)

In [532]:
file.drop(columns=file[['ISO_A3','ISO_A2','ISO_L2','city','type_text','target_other','target_direction','travel_mechanism','update_type','update_level','country','date_updated','recorded_date']],inplace=True)


In [533]:
#just clean the title
file = file.rename(columns={x:x.replace('_', ' ')
    for x in file.columns})
#convert date time
pd.to_datetime(file['date announced'])
pd.to_datetime(file['date start'])
pd.to_datetime(file['date end'])

#extract only month since all study happend in 2020
file['date announced'] = pd.to_datetime(file['date announced']).dt.month
file['date start'] = pd.to_datetime(file['date start']).dt.month
file['date end'] =pd.to_datetime(file['date end']).dt.month
file['date end'] = file['date end'].index.astype(int)

file.head(5)

record id  ... food Transportation
0  R_1IFRA729KAc34j6NA  ...                   0
1  R_1LpM6iIK9Qt3n04NA  ...                   0
2  R_zTscGXXMieVMbehDy  ...                   0
3  R_zTscGXXMieVMbehEa  ...                   0
4  R_zTscGXXMieVMbehFj  ...                   0

[5 rows x 25 columns]

In [534]:
 #print(file.describe().to_string(line_width=60))

now lets play with each column to figure out the values expected ,First fill all nan to o,to not effect the mean .


In [535]:
file['entry type'].value_counts()
#file.target_direction.isnull().value_counts()

new_entry    601
update       589
Name: entry type, dtype: int64

The above tell us the there is 601 policies out of 1190 entered newely and 589 updated from previous one.

In [536]:
#file.fillna(0)

**The column type of correction is perfect column so far ,which has 2 categories org or corr .and we can introduce that the policies has been corrected its effects is mejor than orginal ,so this reveal we can go in deep in 185 policy instead of 1005**

In [537]:
file['init country level'].value_counts()

Provincial    996
National      181
Municipal       6
Name: init country level, dtype: int64

 i decide if the policy be in level of national will effect 3 of them ,but in case of  provinical will just effect farm stabilty and food transportion ,and the effect of municipal will effect only in food transportaion .thats what i conclude and i want to discuss about to continue like this ,based on anlysis fill the table or there is another resourcing can i take to fill the table to start the process

In [538]:
inde=0
for x in file['init country level']:
  inde +=1
  
  if x=='Provincial':
    file.at[inde,'international exchange']=0  
    file.at[inde,'Farm financial stability']=1 
    file.at[inde,'food Transportation']=1
  elif x=='National':
    file.at[inde,'international exchange']=1 
    file.at[inde,'Farm financial stability']=1 
    file.at[inde,'food Transportation']=1
  elif x=='Municipal':
    file.at[inde,'international exchange']=0 
    file.at[inde,'Farm financial stability']=1
    file.at[inde,'food Transportation']=0
  



In [539]:
file.head()

record id  ... food Transportation
0  R_1IFRA729KAc34j6NA  ...                 0.0
1  R_1LpM6iIK9Qt3n04NA  ...                 1.0
2  R_zTscGXXMieVMbehDy  ...                 1.0
3  R_zTscGXXMieVMbehEa  ...                 1.0
4  R_zTscGXXMieVMbehFj  ...                 1.0

[5 rows x 25 columns]

Its time to convert categorical data to nomircal 